persistent/tests/test_persistence.py

```diff
@@ -166,6 +228,14 @@ def _test():
             inst._p_oid = new_OID
         self.assertRaises(ValueError, _test)
 
+    def test_assign_p_oid_not_in_cache_allowed(self):
+        jar = self._makeJar()
+        inst = self._makeOne()
+        inst._p_jar = jar
+        inst._p_oid = 1 # anything goes
+        inst._p_oid = 42
+        self.assertEqual(inst._p_oid, 42)
```

Comment:
> Relying on the fact that the C implementation does not enforce a constraint that is explicit in the interface specification is literally "undocumented behavior" (as with a compiler which may or may not enforce some corner of a language spec tightly). Code which does this is off-roading. 

You've certainly got me there :)

> I could see noting this in a version bump, with a prominent mention on the changelog, if we chose to enforce it.

Alternately, perhaps it would be worth considering relaxing the documentation requirements? That explicitly makes it more useful to people like Stephan, and lets Jim have the freedom to move to UUIDs in the future without having to make changes to this package. Plus it keeps me from having to make more changes to our own testing code :)